jemalloc-sys/src/lib.rs

```diff
@@ -542,6 +542,35 @@ extern "C" {
     #[cfg_attr(prefixed, link_name = "_rjem_malloc_message")]
     pub static mut malloc_message: unsafe extern "C" fn (cbopaque: *mut c_void,
                                                          s: *const c_char);
+
+    /// Compile-time string of configuration options.
+    ///
+    /// Once, when the first call is made to one of the memory allocation
+    /// routines, the allocator initializes its internals based in part on
+    /// various options that can be specified at compile- or run-time.
+    ///
+    /// The string specified via `--with-malloc-conf`, the string pointed to by
+    /// the global variable `malloc_conf`, the “name” of the file referenced by
+    /// the symbolic link named `/etc/malloc.conf`, and the value of the
+    /// environment variable `MALLOC_CONF`, will be interpreted, in that order,
+    /// from left to right as options. Note that `malloc_conf` may be read
+    /// before `main()` is entered, so the declaration of `malloc_conf` should
+    /// specify an initializer that contains the final value to be read by
+    /// `jemalloc`.
+    ///
+    /// `--with-malloc-conf` and `malloc_conf` are compile-time mechanisms, whereas
+    /// `/etc/malloc.conf` and `MALLOC_CONF` can be safely set any time prior to
+    /// program invocation.
+    ///
+    /// An options string is a comma-separated list of `option:value` pairs.
+    /// There is one key corresponding to each `opt.* mallctl` (see the `MALLCTL
+    /// NAMESPACE` section for options documentation). For example,
+    /// `abort:true,narenas:1` sets the `opt.abort` and `opt.narenas` options.
+    /// Some options have boolean values (`true`/`false`), others have integer
+    /// values (base `8`, `10`, or `16`, depending on prefix), and yet others
+    /// have raw string values.
+    #[cfg_attr(prefixed, link_name = "_rjem_malloc_conf")]
+    pub static mut malloc_conf: *const libc::c_char;
```

Comment:
Should this be an `Option<&libc::c_char>` instead ?

Reply:
Is this the right way to expose this? It needs to be set at compile time, but I don't think you can assign to statics before runtime.

Reply:
You can use this to read this, but you cannot use it to assign to it. For how to do that, one of the tests does so using `#[export(name = "")]`.

Reply:
@alexcrichton you mean, like this?

Reply:
Ah yeah, although leaving off the `mut` is fine by me as well

Reply:
So yeah, I'd like that to, but raw pointers are not `Sync`, so if we make it a raw pointer we cannot leave out the `mut`.

OTOH, one cannot mutate this at run-time, so the `mut` does not make sense, and the `static mut` are apparently going to be deprecated anyways, which is why I went with `Option<&T>` =/

Reply:
I could add a `#[repr(transparent)] pub struct CharPtr(*const c_char); unsafe impl Sync for CharPtr {}` and use that instead but that does not feel "great" either :/

Reply:
Oh geez that's quite the mess. Eh in that case I'd just leave it as `Option<&T>`